int32 center
int32 area